src/bson-encode.c

```diff
@@ -143,24 +143,18 @@ static void php_phongo_bson_append_object(bson_t *bson, php_phongo_bson_flags_t
 			if (Z_TYPE(obj_data) != IS_ARRAY && !(Z_TYPE(obj_data) == IS_OBJECT && instanceof_function(Z_OBJCE(obj_data), zend_standard_class_def TSRMLS_CC))) {
 				phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC,
 					"Expected %s::%s() to return an array or stdClass, %s given",
-					Z_OBJCE_P(object)->name->val,
+					ZSTR_VAL(Z_OBJCE_P(object)->name),
 					BSON_SERIALIZE_FUNC_NAME,
-					(Z_TYPE(obj_data) == IS_OBJECT
-						 ? Z_OBJCE(obj_data)->name->val
-						 : zend_get_type_by_const(Z_TYPE(obj_data))
-					)
+					PHONGO_ZVAL_CLASS_OR_TYPE_NAME(obj_data)
 				);
 				zval_ptr_dtor(&obj_data);
 #else
 			if (Z_TYPE_P(obj_data) != IS_ARRAY && !(Z_TYPE_P(obj_data) == IS_OBJECT && instanceof_function(Z_OBJCE_P(obj_data), zend_standard_class_def TSRMLS_CC))) {
 				phongo_throw_exception(PHONGO_ERROR_UNEXPECTED_VALUE TSRMLS_CC,
 					"Expected %s::%s() to return an array or stdClass, %s given",
-					Z_OBJCE_P(object)->name,
+					ZSTR_VAL(Z_OBJCE_P(object)->name),
```

Comment:
Technically, this isn't necessary since we're in PHP 5 mode here, but it reduces some variation between the PHP 5 and 7 blocks. The only real difference in these blocks is that `obj_data` is a `zval *` for PHP 5.